### PR TITLE
Add public API `dataset.get_embeddings()`

### PR DESCRIPTION
**Pull request by dsmilkov**
This leads to a better custom map that uses those embeddings:

```python
ds = ll.get_dataset('local', 'glave-coder-sample')

def get_similarity(x):
  rowid = x[ll.ROWID]
  question_emb = ds.get_embeddings('jina-v2-small', rowid, 'question')[0]['vector']
  answer_emb = ds.get_embeddings('jina-v2-small', rowid, 'answer')[0]['vector']
  return float(np.dot(question_emb, answer_emb))

ds.map(get_similarity, output_column='similarity')
```

Also:
- Show the keyword search option even if the search box has no text yet since a couple of people got confused that you can't search at all unless you compute embeddings
- Fix a bug where we show the embedding badge for all grand-parents up the tree
- Fix a bug with delete signal not working due to urllib encoding the path.
- Fix a bug with uuid replace "-" with "" in a few sources

### CODE REVIEW

File: lilac/data/dataset.py [hunk 1]
```diff
@@ -598,6 +600,26 @@ def add_labels(
     """
     pass
 
+  @abc.abstractmethod
+  def get_embeddings(
+    self, embedding: str, rowid: str, row_path: Union[PathKey, str]
```
Comment (nsthorat) [inline lilac/data/dataset.py:605]: maybe just path? we dont use "row_path" anywhere else

File: lilac/data/dataset_duckdb.py [hunk 2]
```diff
@@ -971,6 +971,21 @@ def _reshard_cache(
...
 
+  @override
+  def get_embeddings(
+    self, embedding: str, rowid: str, row_path: Union[PathKey, str]
+  ) -> list[SpanVector]:
+    """Returns the span-level embeddings associated with a specific row value."""
+    row_path = normalize_path(cast(PathTuple, row_path))
+    row_path = tuple([int(p) if p.isdigit() else p for p in row_path])
+
+    field_path = tuple(['*' if isinstance(p, int) else p for p in row_path])
```
Comment (nsthorat) [inline lilac/data/dataset_duckdb.py:982]: s/'*'/PATH_WILDCARD

File: lilac/data/dataset_test.py [hunk 3]
```diff
@@ -650,3 +650,24 @@ def test_map_dtype(make_test_data: TestDataMaker) -> None:
     {'column': {'key': ['b'], 'value': [2.5]}},
     {'column': {'key': ['a', 'c'], 'value': [3.0, 3.5]}},
   ]
+
+
+def test_get_ebeddings(make_test_data: TestDataMaker) -> None:
```
Comment (nsthorat) [inline lilac/data/dataset_test.py:655]: embeddings

**Review by nsthorat — APPROVED** [review]

File: lilac/data/dataset.py (hunk 1 above)
Comment (dsmilkov) [reply lilac/data/dataset.py:605]: done

File: lilac/data/dataset_test.py (hunk 3 above)
Comment (dsmilkov) [reply lilac/data/dataset_test.py]: done

File: lilac/data/dataset_duckdb.py (hunk 2 above)
Comment (dsmilkov) [reply lilac/data/dataset_duckdb.py:982]: done

**Pull request merged**